pkg/sys/signals_unix.go

```diff
@@ -0,0 +1,22 @@
+// +build !windows,!plan9,!js
+
+package sys
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func NotifySignals() chan os.Signal {
+	// This catches every signal regardless of whether it is ignored.
+	sigCh := make(chan os.Signal, sigsChanBufferSize)
+	signal.Notify(sigCh)
+	// TODO: Remove this if, and when, job control is implemented. This
+	// handles the case of running an external command from an interactive
+	// prompt.
+	//
+	// See https://github.com/elves/elvish/issues/988.
+	signal.Ignore(syscall.SIGTTIN, syscall.SIGTTOU, syscall.SIGTSTP)
```

Comment:
Why is it necessary to do the `Ignore` call here, instead of in `setupShell`?

Reply:
This is because `signal.Notify`, when called without any signals, catches every signal -- even if a signal is currently ignored. So we have to explicitly ignore the TTY signals after calling `signal.Notify` since elvish does not implement UNIX job control. This is done here because this function is called from two locations and might be called twice. Regardless of that happenstance we have to ignore TTY job control signals and do so in any child process that is launched.

Reply:
Interesting, this is something worth documenting in the comment.